management/tenant_test.go

```diff
@@ -11,6 +13,83 @@ func TestTenant(t *testing.T) {
 	var tn *Tenant
 	var err error
 
+	// Based on Auth0 examples, we can verify that the definition of the struct is OK
+	// by unit testing a parse of a json
+	t.Run("should parse all fields", func(t *testing.T) {
```

Comment:
I don't think this is the best way to verify a resource. It verifies json decoding but thats about it. Perhaps we could enable some fields in our tenant and assert whether they are set or not?

Additionally, are these fields documented by auth0? If so it could help me verify.

Reply:
My intention was to validate the json "schema" with a unit test to avoid typos.
I based the name of the fields with this documentation : https://auth0.com/docs/api/management/v2#!/Tenants/get_settings

I think it would be difficult to validate all the fields with an integration test because some fields may be incompatible with each oher.

I also think this simple test can be used as a documentation/exemple for the resource.


But, having said that, if you do not agree, I prefer to align myself with the community's opinion by removing this test.
What do you think ?

Reply:
I understand, it's also not so easy to test all these options as the tenant resource is not one which we can create/delete on demand. 

I'd still prefer to keep things aligned with the rest of the code and perhaps do these types of tests as unit tests in the future.

Reply:
I totally ok with that, but...
...I created the same tests for the Branding resource and the PR has just been merged (sorry).

I propose:
1/ I suppress the tests both in tenant_test and branding_test
2/ I add the same tests for all the resources (in antoher PR)
3/ I add the same tests for all the resources in another file (ex: tenant_u_test) (also in antoher PR)

Choose and I can do it on monday.

Reply:
I choose 1/ 
And I'll propose another PR to de 3/